File: engine/components/writers/src/main/scala/org/datacleaner/beans/writers/WriteDataResultHtmlRenderer.scala
package org.datacleaner.beans.writers
import org.datacleaner.beans.api.Renderer
import org.datacleaner.result.html.HtmlFragment
import org.datacleaner.result.html.HtmlRenderer
import org.datacleaner.result.html.SimpleHtmlFragment
import scala.xml.PrettyPrinter
import javax.xml.transform.Transformer
import org.datacleaner.beans.api.RendererBean
import org.datacleaner.result.renderer.HtmlRenderingFormat
import org.datacleaner.beans.api.RendererPrecedence
import org.datacleaner.configuration.AnalyzerBeansConfiguration
import javax.inject.Inject

@RendererBean(classOf[HtmlRenderingFormat])
class WriteDataResultHtmlRenderer extends Renderer[WriteDataResult, HtmlFragment] {
  
  @Inject
  var configuration: AnalyzerBeansConfiguration = null
  
  override def getPrecedence(renderable: WriteDataResult) = RendererPrecedence.MEDIUM;

  override def render(r: WriteDataResult): HtmlFragment = {
    val inserts = r.getWrittenRowCount()
    val updates = r.getUpdatesCount()
    val errors = r.getErrorRowCount()
    val total = inserts + updates + errors
    val datastoreName = if (configuration == null) null else {
      val ds = r.getDatastore(configuration.getDatastoreCatalog())
      if (ds == null) null else ds.getName()
    }
    
    val html = <div>
                 { if (datastoreName != null) { <p>{if (total == 0) "No data" else "Data"} written to <span class="datastoreName">{datastoreName}</span></p> } }
                 { if (inserts > 0) { <p>Executed { inserts } inserts</p> } }
                 { if (updates > 0) { <p>Executed { updates } updates</p> } }
                 { if (errors > 0) { <p>{ errors } Erroneous records</p> } }
               </div>;

               val frag = new SimpleHtmlFragment();
    frag.addBodyElement(html.toString());
    return frag;
  }
  
  def setConfiguration(configuration: AnalyzerBeansConfiguration) {
    this.configuration = configuration;
  }
}

File: components/writers/src/main/java/org/datacleaner/beans/writers/WriteDataResultHtmlRenderer.java
/**
 * DataCleaner (community edition)
 * Copyright (C) 2014 Neopost - Customer Information Management
 *
 * This copyrighted material is made available to anyone wishing to use, modify,
 * copy, or redistribute it subject to the terms and conditions of the GNU
 * Lesser General Public License, as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
 * for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this distribution; if not, write to:
 * Free Software Foundation, Inc.
 * 51 Franklin Street, Fifth Floor
 * Boston, MA  02110-1301  USA
 */
package org.datacleaner.beans.writers;

import javax.inject.Inject;

import org.datacleaner.api.Renderer;
import org.datacleaner.api.RendererBean;
import org.datacleaner.api.RendererPrecedence;
import org.datacleaner.configuration.DataCleanerConfiguration;
import org.datacleaner.connection.Datastore;
import org.datacleaner.result.html.HtmlFragment;
import org.datacleaner.result.html.SimpleHtmlFragment;
import org.datacleaner.result.renderer.HtmlRenderingFormat;

@RendererBean(HtmlRenderingFormat.class)
public class WriteDataResultHtmlRenderer implements Renderer<WriteDataResult, HtmlFragment> {

    @Inject
    DataCleanerConfiguration _configuration;

    @Override
    public RendererPrecedence getPrecedence(WriteDataResult renderable) {
        return RendererPrecedence.MEDIUM;
    }

    public void setConfiguration(DataCleanerConfiguration configuration) {
        _configuration = configuration;
    }

    @Override
    public HtmlFragment render(WriteDataResult r) {
        final int inserts = r.getWrittenRowCount();
        final int updates = r.getUpdatesCount();
        final int errors = r.getErrorRowCount();
        final int total = inserts + updates + errors;
        final Datastore datastore = (_configuration == null ? null : r.getDatastore(_configuration
                .getDatastoreCatalog()));

        final StringBuilder sb = new StringBuilder("<div>");
        if (datastore != null) {
            sb.append("\n  <p>");
            if (total == 0) {
                sb.append("No data");
            } else {
                sb.append("Data");
            }
            sb.append(" written to <span class=\"datastoreName\">");
            sb.append(datastore.getName());
            sb.append("</span></p>");
        }

        if (inserts > 0) {
            sb.append("\n  <p>Executed ");
            sb.append(inserts);
            sb.append(" inserts</p>");
        }

        if (updates > 0) {
            sb.append("\n  <p>Executed ");
            sb.append(updates);
            sb.append(" updates</p>");
        }

        if (errors > 0) {
            sb.append("\n  <p>");
            sb.append(errors);
            sb.append(" Erroneous records</p>");
        }

        sb.append("\n</div>");

        final SimpleHtmlFragment frag = new SimpleHtmlFragment();
        frag.addBodyElement(sb.toString());
        return frag;
    }
}
